doc/en/user/source/extensions/css/workshop/linestring.rst

```diff
@@ -0,0 +1,500 @@
+
+   .. image:: img/line_05_scale.png
+
+#. Putting these two ideas together allows control of level detail based on scale:
+
+   .. code-block:: css
+
+      [@scale < 9000000] [scalerank > 7] {
+        stroke: #888888;
+        stroke-width: 2;
+      }
+      [@scale > 9000000] [@scale < 17000000] [scalerank < 7] {
+        stroke: #777777;
+      }
+      [@scale > 1700000] [@scale < 35000000] [scalerank < 6] {
+        stroke: #444444;
+      }
+      [@scale > 3500000] [@scale < 70000000] [scalerank < 5] {
+        stroke: #000055;
+      }
+      [@scale > 70000000] [scalerank < 4] {
+        stroke: black;
+      }
+
+#. As shown above selectors can be combined in the same rule:
+
+   * Selectors separated by whitespace are combined CQL Filter AND
+   * Selectors separated by a comma are combined using CQL Filter OR
+
+   Our first rule `[@scale < 9000000] [scalerank > 7]` checks that the scale is less than 9M AND scalerank is greater than 7.
+
+   .. image:: img/line_06_adjust.png
+   
+
+Additional Considerations
+-------------------------
+
+Vendor options can be used to enable some quite spectacular effects, while still providing a style that can be used by other applications.
+
+#. Update `line_example` with the following:
+
+   .. code-block:: css
+
+       * {
+         stroke: ededff;
+         stroke-width: 10;
+         label: [level] " " [name];
+         font-fill: black;
+         -gt-label-follow-line: true;
+       }
+
+#. The property **stroke-width** has been used to make our line thicker in order to provide a backdrop for our label. 
+
+   .. code-block:: css
+       :emphasize-lines: 3
+    
+       * {
+         stroke: ededff;
+         stroke-width: 10;
+         label: [level] " " [name];
+         font-fill: black;
+         -gt-label-follow-line: true;
+       }
+
+#. The **label** property combines combine several CQL expressions together for a longer label.
+
+   .. code-block:: css
+       :emphasize-lines: 4
+
+       * {
+         stroke: ededff;
+         stroke-width: 10;
+         label: [level] " " [name];
+         font-fill: black;
+         -gt-label-follow-line: true;
+       }
+
+   The combined **label** property::
+       
+       [level] " " [name]
+       
+   Is internally represented with the **Concatenate** function::
+
+       [Concatenate(level,' #', name)] 
+
+#. The property **-gt-label-follow-line** provides the ability of have a label exactly follow a LineString character by character.
+
+   .. code-block:: css
+       :emphasize-lines: 6
+    
+       * {
+         stroke: ededff;
+         stroke-width: 10;
+         label: [level] " " [name];
+         font-fill: black;
+         -gt-label-follow-line: true;
+       }
+
+#. The result is a new appearance for our roads.
+
+   .. image:: img/line_label_4.png
+
+#. Generate the SLD for the following CSS.
```

Comment:
Please put each Challenge inside notes or other admonitions, or separate them in some way. They specifically need to be blocked out from the normal flow of the text, otherwise they disrupt the flow of the narrative.